tests/js/spec/views/eventsV2/eventView.spec.jsx

```diff
@@ -368,6 +368,202 @@ describe('EventView.fromSavedQuery()', function() {
   });
 });
 
+describe('EventView.fromNewQueryWithLocation()', function() {
+  const prebuiltQuery = {
+    id: undefined,
+    name: 'All Events',
+    query: '',
+    projects: [],
+    fields: ['title', 'event.type', 'project', 'user', 'timestamp'],
+    fieldnames: ['title', 'type', 'project', 'user', 'time'],
+    orderby: '-timestamp',
+    version: 2,
+    tags: [
+      'event.type',
+      'release',
+      'project.name',
+      'user.email',
+      'user.ip',
+      'environment',
+    ],
+  };
+
+  it('maps basic properties of a prebuilt query', function() {
+    const location = {
+      query: {
+        statsPeriod: '99d',
+      },
+    };
+
+    const eventView = EventView.fromNewQueryWithLocation(prebuiltQuery, location);
+
+    expect(eventView).toMatchObject({
+      id: undefined,
+      name: 'All Events',
+      fields: [
+        {field: 'title', title: 'title'},
+        {field: 'event.type', title: 'type'},
+        {field: 'project', title: 'project'},
+        {field: 'user', title: 'user'},
+        {field: 'timestamp', title: 'time'},
+      ],
+      sorts: [{field: 'timestamp', kind: 'desc'}],
+      tags: [
+        'event.type',
+        'release',
+        'project.name',
+        'user.email',
+        'user.ip',
+        'environment',
+      ],
+      query: '',
+      project: [],
+      start: undefined,
+      end: undefined,
+      // statsPeriod has precedence
+      statsPeriod: '99d',
+      environment: [],
+      yAxis: undefined,
+    });
+  });
+
+  it('merges global selection values', function() {
+    const location = {
+      query: {
+        statsPeriod: '99d',
+        project: ['456'],
+        environment: ['prod'],
+      },
+    };
+
+    const eventView = EventView.fromNewQueryWithLocation(prebuiltQuery, location);
+
+    expect(eventView).toMatchObject({
+      id: undefined,
+      name: 'All Events',
+      fields: [
+        {field: 'title', title: 'title'},
+        {field: 'event.type', title: 'type'},
+        {field: 'project', title: 'project'},
+        {field: 'user', title: 'user'},
+        {field: 'timestamp', title: 'time'},
+      ],
+      sorts: [{field: 'timestamp', kind: 'desc'}],
+      tags: [
+        'event.type',
+        'release',
+        'project.name',
+        'user.email',
+        'user.ip',
+        'environment',
+      ],
+      query: '',
+      project: [456],
+      start: undefined,
+      end: undefined,
+      statsPeriod: '99d',
+      environment: ['prod'],
+      yAxis: undefined,
+    });
+  });
+
+  it('new query takes precedence over global selection values', function() {
```

Comment:
Could this result in the global header saying one thing but the result set being for a different range? How do we sync those up?

Reply:
@markstory  Good question. Right now, I've meant `EventView.fromNewQueryWithLocation()` to be only used for pre-built queries; which at the moment, do not have `statsPeriod` baked in. But I've written code in the case if a decision was made to bake `statsPeriod` in the upcoming weeks.

This merging strategy is used for the "Build a new query" CTA, and as well as for the query cards CTA for pre-built queries (and their minigraphs).

This is primarily for the queries we display on the landing page; since the global header is accessible to the user on this page. In other words, changing the values on the global header should be reflected to the minigraphs on the prebuilt query cards.

Regarding your concerns in sync'ing of the global header and the query, we already ensure that `EventView.fromLocation()` is the source of truth for the query builder view.

-------

For saved queries, any values they may have will override the global selection values (this is what `EventView.fromSavedQuery()` does).